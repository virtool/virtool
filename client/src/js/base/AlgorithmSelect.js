import React from "react";
import PropTypes from "prop-types";
import { map } from "lodash-es";

import { Input } from "./";
import { getTaskDisplayName } from "../utils";

export const algorithms = [
    "pathoscope_bowtie",
    "nuvs"
];

/**
 * An input-based component for selecting an algorithm (eg. pathoscope_bowtie), by its display name
 * (eg. Pathoscope Bowtie).
 *
 */
export const AlgorithmSelect = (props) => (
    <Input
        name="algorithm"
        type="select"
        label={props.noLabel ? null : "Algorithm"}
        value={props.value}
        onChange={props.onChange}
    >
        {map(algorithms, algorithm =>
            <option
                key={algorithm}
                value={algorithm}
                disabled={(algorithm === "nuvs" && !props.hasHmm)}
            >
                {getTaskDisplayName(algorithm)}
            </option>
        )}
    </Input>
);

AlgorithmSelect.propTypes = {
    noLabel: PropTypes.bool,
    value: PropTypes.oneOf(["pathoscope_bowtie", "nuvs"]),
    onChange: PropTypes.func,
    hasHmm: PropTypes.bool
};

