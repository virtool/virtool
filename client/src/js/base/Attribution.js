import { capitalize } from "lodash-es";
import React from "react";
import styled from "styled-components";
import { InitialIcon } from "./InitialIcon";
import { RelativeTime } from "./RelativeTime";

export const UnstyledAttribution = ({ className, time, user, verb = "created" }) => {
    return (
        <span className={className}>
            {user ? <InitialIcon size="md" handle={user} /> : null}
            <span>{user}</span>
            <span>{user ? verb : capitalize(verb)}</span>
            <RelativeTime time={time} />
        </span>
    );
};

export const Attribution = styled(UnstyledAttribution)`
    align-items: center;
    display: inline-flex;
    font-size: inherit;

    *:not(:first-child) {
        margin-left: 3px;
    }
    .InitialIcon {
        margin-right: 5px;
    }
`;
