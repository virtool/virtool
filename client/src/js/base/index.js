import { AlgorithmSelect } from "./AlgorithmSelect";
import { AutoProgressBar, ProgressBar } from "./ProgressBar";
import { bsStyles } from "./utils";
import { Checkbox } from "./Checkbox";
import { Flex, FlexItem } from "./Flex";
import { Help } from "./Help";
import { Icon } from "./Icon";
import { Identicon } from "./Identicon";
import { Input } from "./Input";
import { InputError } from "./InputError";
import { InputSave } from "./InputSave";
import { ListGroupItem } from "./ListGroupItem";
import { NotFound } from "./NotFound";
import { Panel } from "./Panel";
import { Radio } from "./Radio";
import { RelativeTime } from "./RelativeTime";
import { RemoveBanner } from "./RemoveBanner";
import { RemoveModal } from "./RemoveModal";
import { SaveButton } from "./SaveButton";
import { ScrollList } from "./ScrollList";
import { Tooltip } from "./Tooltip";
import { UploadBar } from "./UploadBar";
import { ViewHeader } from "./ViewHeader";

export * from "./Alert";
export * from "./Attribution";
export * from "./Badge";
export * from "./Box";
export * from "./Breadcrumb";
export * from "./Breakpoints";
export * from "./Button";
export * from "./ButtonGroup";
export * from "./ButtonToolbar";
export * from "./Code";
export * from "./Container";
export * from "./ExternalLink";
export * from "./Label";
export * from "./Loader";
export * from "./LoadingPlaceholder";
export * from "./Logo";
export * from "./Modal";
export * from "./NoneFound";
export * from "./Table";
export * from "./SectionHeader";
export * from "./SlashList";
export * from "./TabLink";
export * from "./Tabs";
export * from "./Toolbar";
export * from "./ViewHeader";
export * from "./SelectBox";

export {
    AlgorithmSelect,
    AutoProgressBar,
    bsStyles,
    Checkbox,
    Flex,
    FlexItem,
    Help,
    Icon,
    Identicon,
    Input,
    InputError,
    InputSave,
    ListGroupItem,
    NotFound,
    Panel,
    ProgressBar,
    Radio,
    RelativeTime,
    RemoveBanner,
    RemoveModal,
    SaveButton,
    ScrollList,
    Tooltip,
    UploadBar,
    ViewHeader
};
