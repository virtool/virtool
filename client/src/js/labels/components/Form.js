import React from "react";
import { Button, Input, InputError, InputGroup, InputLabel } from "../../base";
import { ColorSelector } from "./ColorSelector";

export const LabelForm = ({ color, description, errorColor, errorName, name, onChange, onColorChange, onSubmit }) => (
    <form onSubmit={onSubmit}>
        <InputGroup>
            <InputLabel>Name</InputLabel>
            <Input name="labelName" value={name} onChange={onChange} error={errorName} />
            <InputError>{errorName}</InputError>
            <InputLabel>Description</InputLabel>
            <Input name="description" value={description} onChange={onChange} />
        </InputGroup>
        <ColorSelector color={color} errorColor={errorColor} onColorChange={onColorChange} />
        <Button type="submit" color="blue">
            Create
        </Button>
    </form>
);
