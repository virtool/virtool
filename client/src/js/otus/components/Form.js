import React from "react";
import styled from "styled-components";
import { ModalBody, ModalFooter, Input, InputError, InputGroup, InputLabel, SaveButton } from "../../base";

const OTUFormBody = styled(ModalBody)`
    display: grid;
    grid-template-columns: 9fr 4fr;
    grid-column-gap: ${props => props.theme.gap.column};
`;

export const OTUForm = ({ abbreviation, name, error, onChange, onSubmit }) => (
    <form onSubmit={onSubmit}>
        <OTUFormBody>
            <InputGroup>
                <InputLabel>Name</InputLabel>
                <Input error={error} name="name" value={name} onChange={onChange} />
                <InputError>{error}</InputError>
            </InputGroup>

            <InputGroup>
                <InputLabel>Abbreviation</InputLabel>
                <Input name="abbreviation" value={abbreviation} onChange={onChange} />
            </InputGroup>
        </OTUFormBody>
        <ModalFooter>
            <SaveButton />
        </ModalFooter>
    </form>
);
