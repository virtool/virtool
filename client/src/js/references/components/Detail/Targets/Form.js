import React from "react";
import { Checkbox, Input, InputError, InputGroup, InputLabel, TextArea } from "../../../../base";

export const TargetForm = ({ description, errorName, length, name, required, onChange, onClick }) => {
    return (
        <React.Fragment>
            <InputGroup>
                <InputLabel>Name</InputLabel>
                <Input error={errorName} name="name" value={name} onChange={onChange} />
                <InputError>{errorName}</InputError>
            </InputGroup>

            <InputGroup>
                <InputLabel>Description</InputLabel>
                <TextArea name="description" value={description} onChange={onChange} />
            </InputGroup>

            <InputGroup>
                <InputLabel>Length</InputLabel>
                <Input type="number" name="length" value={length} onChange={onChange} />
            </InputGroup>

            <Checkbox label="Required" checked={required} onClick={onClick} />
        </React.Fragment>
    );
};
