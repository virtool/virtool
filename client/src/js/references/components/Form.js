import React from "react";
import { Input, InputError, InputGroup, InputLabel, TextArea } from "../../base";

export const ReferenceForm = ({ description, organism, mode, name, errorName, onChange }) => {
    let organismComponent;

    if (mode === "empty") {
        organismComponent = (
            <InputGroup>
                <InputLabel>Organism</InputLabel>
                <Input name="organism" value={organism} onChange={onChange} />
            </InputGroup>
        );
    }

    return (
        <React.Fragment>
            <InputGroup>
                <InputLabel>Name</InputLabel>
                <Input name="name" error={errorName} value={name} onChange={onChange} />
                <InputError>{errorName}</InputError>
            </InputGroup>

            {organismComponent}

            <InputGroup>
                <InputLabel>Description</InputLabel>
                <TextArea name="description" value={description} onChange={onChange} />
            </InputGroup>
        </React.Fragment>
    );
};
