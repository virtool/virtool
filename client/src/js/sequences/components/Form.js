import React from "react";
import PropTypes from "prop-types";
import { Input, InputContainer, InputError, InputGroup, InputLabel } from "../../base";
import { getTargetChange } from "../../utils/utils";
import { Accession } from "./Accession";
import SequenceField from "./Field";

export const SequenceForm = ({ accession, definition, errors, host, sequence, onChange }) => {
    const handleAutofill = ({ accession, definition, host, sequence }) =>
        onChange({
            accession,
            definition,
            host,
            sequence
        });

    const handleChange = e => {
        const { name, value } = getTargetChange(e.target);
        onChange({ [name]: value });
    };

    return (
        <React.Fragment>
            <Accession
                accession={accession}
                error={errors.accession}
                onAutofill={handleAutofill}
                onChange={handleChange}
            />

            <InputGroup>
                <InputLabel>Host</InputLabel>
                <Input name="host" value={host} onChange={handleChange} />
            </InputGroup>

            <InputGroup>
                <InputLabel>Definition</InputLabel>
                <InputContainer>
                    <Input name="definition" value={definition} onChange={handleChange} />
                    <InputError>{errors.definition}</InputError>
                </InputContainer>
            </InputGroup>

            <SequenceField name="sequence" sequence={sequence} onChange={handleChange} error={errors.sequence} />
        </React.Fragment>
    );
};

SequenceForm.propTypes = {
    accession: PropTypes.string.isRequired,
    definition: PropTypes.string.isRequired,
    errors: PropTypes.object.isRequired,
    host: PropTypes.string.isRequired,
    sequence: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired
};
