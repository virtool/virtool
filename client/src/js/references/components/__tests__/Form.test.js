import { Input, TextArea } from "../../../base";
import { ReferenceForm } from "../Form";

describe("<ReferenceForm />", () => {
    let props;

    beforeEach(() => {
        props = {
            description: "Foo reference",
            errorName: "baz",
            mode: "clone",
            name: "Foo",
            organism: "Bar",
            onChange: jest.fn()
        };
    });

    it("should render", () => {
        const wrapper = shallow(<ReferenceForm {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render error when provided in props", () => {
        props.errorName = "Name required";
        const wrapper = shallow(<ReferenceForm {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render organism field when [mode=empty]", () => {
        props.mode = "empty";
        const wrapper = shallow(<ReferenceForm {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it.each(["name", "organism", "description"])("should call onChange() when %p input changes", name => {
        props.mode = "empty";
        const wrapper = shallow(<ReferenceForm {...props} />);
        const e = { target: { name, value: "Baz" } };
        wrapper
            .find(name === "description" ? TextArea : Input)
            .at(name === "organism" ? 1 : 0)
            .simulate("change", e);
        expect(props.onChange).toHaveBeenCalledWith(e);
    });
});
