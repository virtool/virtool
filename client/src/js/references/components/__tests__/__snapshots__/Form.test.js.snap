// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReferenceForm /> should render 1`] = `
<Fragment>
  <InputGroup>
    <Input__InputLabel>
      Name
    </Input__InputLabel>
    <Input
      error="baz"
      name="name"
      onChange={[MockFunction]}
      value="Foo"
    />
    <Input__InputError>
      baz
    </Input__InputError>
  </InputGroup>
  <InputGroup>
    <Input__InputLabel>
      Description
    </Input__InputLabel>
    <TextArea
      name="description"
      onChange={[MockFunction]}
      value="Foo reference"
    />
  </InputGroup>
</Fragment>
`;

exports[`<ReferenceForm /> should render error when provided in props 1`] = `
<Fragment>
  <InputGroup>
    <Input__InputLabel>
      Name
    </Input__InputLabel>
    <Input
      error="Name required"
      name="name"
      onChange={[MockFunction]}
      value="Foo"
    />
    <Input__InputError>
      Name required
    </Input__InputError>
  </InputGroup>
  <InputGroup>
    <Input__InputLabel>
      Description
    </Input__InputLabel>
    <TextArea
      name="description"
      onChange={[MockFunction]}
      value="Foo reference"
    />
  </InputGroup>
</Fragment>
`;

exports[`<ReferenceForm /> should render organism field when [mode=empty] 1`] = `
<Fragment>
  <InputGroup>
    <Input__InputLabel>
      Name
    </Input__InputLabel>
    <Input
      error="baz"
      name="name"
      onChange={[MockFunction]}
      value="Foo"
    />
    <Input__InputError>
      baz
    </Input__InputError>
  </InputGroup>
  <InputGroup>
    <Input__InputLabel>
      Organism
    </Input__InputLabel>
    <Input
      name="organism"
      onChange={[MockFunction]}
      value="Bar"
    />
  </InputGroup>
  <InputGroup>
    <Input__InputLabel>
      Description
    </Input__InputLabel>
    <TextArea
      name="description"
      onChange={[MockFunction]}
      value="Foo reference"
    />
  </InputGroup>
</Fragment>
`;
